### ISSUE TITLE
bug(datepicker):  Header row cells have low color contrast ratio

### ISSUE THREAD

**Issue by zarend** [angular/components#23511]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the “Basic date range picker” demo
3. Open the calendar dialog
4. Observe the contrast on the day of the week labels (S M T W T F S).
 
 
#### Expected Behavior
It should have a contrast ratio of at least 4.5:1.

#### Actual Behavior
It has a contrast ratio of 2.67.

#### Environment
OS| Browser |
ChromeOS | Google Chrome 91.0.4472.167 |
macOS Catalina | Google Chrome 91.0.4472.164 |
Windows 10 | Google Chrome 91.0.4472.106 |

##### Additional Notes
This is copied from an [internal bug report](http://b/195151957) found during a11y testing.
